# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/linux/hotfix-dreamcast-bba.patch.superh
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

It looks like the dreamcast network adapter (BBA) rtl8139too doesn't work in
PIO mode anymore, so the claimed memory apparently needs to be of type
IORESOURCE_MEM.

If it is flagged IORESOURCE_IO, the msg at boot is and leaves ethernet defunct:

8139too: 8139too Fast Ethernet driver 0.9.28
8139too 0000:00:00.0: BAR 1 [mem 0x01001700-0x010018ff]: not claimed; can't enable device
resource: Trying to free nonexistent resource <0x01001800-0x010018ff>
resource: Trying to free nonexistent resource <0x01001700-0x010018ff>
8139too 0000:00:00.0: probe with driver 8139too failed with error -22

Speed iperf between dreamcast and PC:
[ ID] Interval           Transfer     Bitrate         Retr
[  5]   0.00-10.00  sec  21.0 MBytes  17.6 Mbits/sec  977             sender
[  5]   0.00-10.01  sec  20.9 MBytes  17.5 Mbits/sec                  receiver

 -- foxdrodd (fuchsfl at gmail.com)

--- linux-6.15.1/arch/sh/drivers/pci/pci-dreamcast.c.vanilla	2025-06-04 14:46:27.000000000 +0200
+++ linux-6.15.1/arch/sh/drivers/pci/pci-dreamcast.c	2025-06-08 23:46:42.520794933 +0200
@@ -27,7 +27,7 @@
 		.name	= "GAPSPCI IO",
 		.start	= GAPSPCI_BBA_CONFIG,
 		.end	= GAPSPCI_BBA_CONFIG + GAPSPCI_BBA_CONFIG_SIZE - 1,
-		.flags	= IORESOURCE_IO,
+		.flags	= IORESOURCE_MEM,
 	},  {
 		.name	= "GAPSPCI mem",
 		.start	= GAPSPCI_DMA_BASE,
